### PR TITLE
Configure eslint and prettiier

### PR DESCRIPTION
**Pull request by technophile-04**
### Description : 

- Configured prettier inline with SE-2 repo config
- Formatted files with prettier
- Configured [ESlint v9](https://eslint.org/docs/latest/use/getting-started) and [typescript-eslint](https://typescript-eslint.io/getting-started/)
- Both prettier and eslint ignores templates dir
- Fixed eslint issues
- Configured pre-commit hooks with husky and lint-staged

Tried keeping typescript-eslint rules inline with SE-2. 

## To test : 
1. Install dependencies : 
```
yarn install
```

2. Add `const x = 5` in `src/cli.ts` and try running : 
```
git add . && git commit -m "this should fail"
```

This will run pre-commit and hopefully it should fail (since x is unused variable) 

cc @Pabl0cks, could you please test this on windows also once, because I remmember in SE-2 while configuring husky it was not working properly especially for windows. 

## New Commands : 

To run lint : 

```
yarn lint
```

or to fix all fixable errors:  `yarn lint --fix`

To format files with prettier : 

```
yarn format
```

### CODE REVIEW

File: src/tasks/copy-template-files.ts [hunk 1]
```diff
@@ -10,134 +10,99 @@ import ncp from "ncp";
...
   return expandedExtensions;
 };
 
-const isTemplateRegex = /([^\/\\]*?)\.template\./;
+const isTemplateRegex = /([^/\\]*?)\.template\./;
 const isPackageJsonRegex = /package\.json/;
 const isYarnLockRegex = /yarn\.lock/;
-const isNextGeneratedRegex = /packages\/nextjs\/generated/
-const isConfigRegex = /([^\/\\]*?)\\config\.json/;
-const isArgsRegex = /([^\/\\]*?)\.args\./;
```
Comment (technophile-04) [inline src/tasks/copy-template-files.ts:41]: Screenshot 2024-05-23 at 5 19 18 PM


Lol, it not best practice to touch working regex and I could have added eslint ignore comment about them 

But removing the backslash made sense checkout [this](https://chatgpt.com/share/eb379c4f-a663-4c87-bae4-8b06bab520c4), also tested it nicely and works nicely

**Comment by carletex** [comment]
This is great @technophile-04 , thank you!! I feel cleaner already haha

Everything works great, except Husky. It's not working for me (I tried deleting node_modules and also doing a fresh install in a new folder). Is it because we are missing these scripts in package.json??
```
    "postinstall": "husky install",
    "precommit": "lint-staged",
```?

I see you added the `prepare` one, but doesn't seem to be working for me.

**Comment by rin-st** [comment]
I tried to commit using vscode ui button and it didn't work. But after running `git add . && git commit -m "this should fail"` in cli it started to work (without `"precommit": "lint-staged",`)

looks like it runs script from `.linstagedrc.js` on commit

**Comment by carletex** [comment]
image

Started to work for me too... but not sure how it happened haha I always use the CLI (not my IDE) for git stuff. But somehow, tried again and now it's showing the lintstaged stuff 🤷

**Comment by carletex** [comment]
Tried in a fresh clone (clone + yarn install + tweak cli.ts + git add + git commit) and not working

image

**Comment by technophile-04** [comment]
Just pushed [2df5d82](https://github.com/scaffold-eth/create-eth/pull/35/commits/2df5d82d020cff187d99983e983def48b0692213),  could you all please try again 

Yarn v3 does not support "prepare" script, so we need to use "postinstall" as mentioned https://github.com/typicode/husky/pull/1371

notice I didn't used "husky install" instead used just "husky" since v9 they updated this 

I just tried clone + yarn install + tweak cli.ts + git add + git commit and it worked in first try

**Comment by rin-st** [comment]
> I just tried clone + yarn install + tweak cli.ts + git add + git commit and it worked in first try

Thanks! It worked for me too

**Review by rin-st — APPROVED** [review]
Great job!

**Comment by carletex** [comment]
Working for me too!! Thanks.

Let's wait until @Pabl0cks can do a quick run on Windows.

**Review by carletex — APPROVED** [review]

**Review by damianmarti — APPROVED** [review]
Working fine!!

/ssd/workspace/create-eth/src/cli.ts
  8:7  error  'x' is assigned a value but never used  @typescript-eslint/no-unused-vars

✖ 1 problem (1 error, 0 warnings)

Thanks!!

**Review by Pabl0cks — APPROVED** [review]
All commands worked nicely on Windows, GJ @technophile-04 

And got the error nicely when trying to commit:

image

**Comment by technophile-04** [comment]
Thanks all ! Merging this 🙌

**Pull request merged**